### PR TITLE
Enable endpoint customization for each integration

### PR DESCRIPTION
**Pull request by maciejwalkowiak**
## :loudspeaker: Type of change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring

## :scroll: Description

Adds the ability to specify custom endpoint on service integration properties making it very easy to configure Spring Cloud AWS with Localstack without a need to overwrite a client bean.

## :bulb: Motivation and Context

#570 

## :green_heart: How did you test it?

Unit & Integration tests.

## :pencil: Checklist

- [x] I reviewed submitted code
- [x] I added tests to verify changes
- [ ] I updated reference documentation to reflect the change
- [x] All tests passing
- [x] No breaking changes

### CODE REVIEW

**Comment by maciejwalkowiak** [comment]
Docs will be updated soon.

File: spring-cloud-aws-autoconfigure/src/main/java/org/springframework/cloud/aws/autoconfigure/jdbc/AmazonRdsDatabaseAutoConfiguration.java [hunk 1]
```diff
@@ -77,10 +77,13 @@ public static RdsInstanceConfigurerBeanPostProcessor rdsInstanceConfigurerBeanPo
...
 				BeanDefinitionRegistry registry) {
-			String amazonRdsClientBeanName = AmazonWebserviceClientConfigurationUtils.registerAmazonWebserviceClient(
-					this, registry, "com.amazonaws.services.rds.AmazonRDSClient", null, null).getBeanName();
-
-			rdsDatabaseProperties().getInstances().stream().filter(RdsInstance::hasRequiredPropertiesSet)
+			AmazonRdsDatabaseProperties properties = rdsDatabaseProperties();
+			String amazonRdsClientBeanName = AmazonWebserviceClientConfigurationUtils
+					.registerAmazonWebserviceClient(this, registry, "com.amazonaws.services.rds.AmazonRDSClient", null,
+							properties.getRegion(),
+							properties.getEndpoint() != null ? properties.getEndpoint().toString() : null)
```
Comment (eddumelendez) [inline spring-cloud-aws-autoconfigure/src/main/java/org/springframework/cloud/aws/autoconfigure/jdbc/AmazonRdsDatabaseAutoConfiguration.java:84]: can we move this to `String endpoint = properties.getEndpoint() != null ? properties.getEndpoint().toString() : null;` and then use endpoint here?

File: spring-cloud-aws-autoconfigure/src/main/resources/META-INF/additional-spring-configuration-metadata.json [hunk 2]
```diff
@@ -47,6 +47,18 @@
...
-    }
+    },
+	{
+	  "defaultValue": null,
+	  "name": "cloud.aws.s3.region",
+	  "description": "Overrides the default region.",
+	  "type": "java.lang.String"
+	},
+	{
+	  "defaultValue": null,
```
Comment (eddumelendez) [inline spring-cloud-aws-autoconfigure/src/main/resources/META-INF/additional-spring-configuration-metadata.json:58]: can be removed

File: spring-cloud-aws-autoconfigure/src/main/resources/META-INF/additional-spring-configuration-metadata.json [hunk 3]
```diff
@@ -47,6 +47,18 @@
       "name": "cloud.aws.stack.enabled",
       "description": "Enables Stack integration.",
       "type": "java.lang.Boolean"
-    }
+    },
+	{
+	  "defaultValue": null,
```
Comment (eddumelendez) [inline spring-cloud-aws-autoconfigure/src/main/resources/META-INF/additional-spring-configuration-metadata.json:52]: can be removed

File: spring-cloud-aws-core/src/main/java/org/springframework/cloud/aws/core/config/AwsClientProperties.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+package org.springframework.cloud.aws.core.config;
+
+import java.net.URI;
+
+/**
+ * Base properties for AWS Service client.
+ *
+ * @author Maciej Walkowiak
+ */
+public abstract class AwsClientProperties {
```
Comment (eddumelendez) [inline spring-cloud-aws-core/src/main/java/org/springframework/cloud/aws/core/config/AwsClientProperties.java:26]: I like this but I think we could create a PR just introducing this change and later a new PR introducing the endpoint property.

**Review by eddumelendez — CHANGES_REQUESTED** [review]
nice feature!

**Comment by maciejwalkowiak** [comment]
@eddumelendez I added short docs section. I did not get into details on how to set up with Localstack. Perhaps later.

**Pull request merged**